clang-tools-extra/clang-tidy/readability/UseCppStyleCommentsCheck.cpp

```diff
@@ -0,0 +1,194 @@
+
+      if (NewlinePos != StringRef::npos) {
+        CommentText = CommentText.substr(NewlinePos + 1);
+        LineNo++;
+      } else {
+        CommentText = "";
+      }
+    }
+    while (!CommentText.empty()) {
+      const size_t NewlinePos = CommentText.find('\n');
+
+      StringRef Line = (NewlinePos == StringRef::npos)
+                           ? CommentText
+                           : CommentText.substr(0, NewlinePos);
+
+      const SourceLocation LineStart = SM.translateLineCol(FID, LineNo, 1);
+      const SourceLocation LineEnd = SM.translateLineCol(
+          FID, LineNo, std::numeric_limits<unsigned>::max());
+
+      if (Line.substr(0, 2) == "  ") {
+        Line = Line.drop_front(2);
+        FixIts.push_back(FixItHint::CreateReplacement(
+            CharSourceRange::getCharRange(LineStart, LineEnd),
+            "//" + Line.str()));
+      } else {
+        FixIts.push_back(FixItHint::CreateReplacement(
+            CharSourceRange::getCharRange(LineStart, LineEnd),
+            "//" + Line.str()));
+      }
+
+      if (NewlinePos == StringRef::npos) {
+        break;
+      }
+      CommentText = CommentText.substr(NewlinePos + 1);
+      LineNo++;
+    }
+  }
+
+  bool isDoxygenStyleComment(const StringRef &Text) {
+    StringRef Trimmed = Text.ltrim();
+    return Trimmed.starts_with("/**") || Trimmed.starts_with("/*!") ||
+           Trimmed.starts_with("///") || Trimmed.starts_with("//!") ||
+           (Trimmed.starts_with("/*") &&
+            Trimmed.drop_front(2).starts_with("*"));
+  }
+
+  bool CheckForCodeAfterComment(Preprocessor &PP, SourceRange Range) {
+    const SourceManager &SM = PP.getSourceManager();
+    const SourceLocation CommentStart = Range.getBegin(),
+                         CommentEnd = Range.getEnd();
+    const std::optional<Token> NextTok =
+        Lexer::findNextToken(CommentStart, SM, PP.getLangOpts());
+    if (!NextTok.has_value()) {
+      return false;
+    }
+    const std::string tokenSpelling =
+        Lexer::getSpelling(*NextTok, SM, PP.getLangOpts());
+    const unsigned lineNo = SM.getSpellingLineNumber(CommentEnd);
+    const SourceLocation loc = NextTok->getLocation();
+    const unsigned tokenLine = SM.getSpellingLineNumber(loc);
+    return lineNo == tokenLine;
+  }
+
+  bool HandleComment(Preprocessor &PP, SourceRange Range) override {
+    const SourceManager &SM = PP.getSourceManager();
+
+    if (Range.getBegin().isMacroID() || SM.isInSystemHeader(Range.getBegin())) {
+      return false;
+    }
+
+    const StringRef Text = Lexer::getSourceText(
+        CharSourceRange::getCharRange(Range), SM, PP.getLangOpts());
+
+    if (!ExcludedCommentMatch.isValid()) {
+      llvm::errs() << "Warning: Invalid regex pattern:" << ExcludedComments
+                   << ":for ExcludedComments\n";
+    } else if (ExcludedCommentMatch.match(Text)) {
+      return false;
+    }
+    if (ExcludeDoxygen && isDoxygenStyleComment(Text)) {
+      return false;
+    }
+
```

Comment:
You can ignore brace if there are only one line inside. It is default code format for LLVM. please fix all similar cases.
```suggestion
    if (ExcludeDoxygen && isDoxygenStyleComment(Text))
      return false;

```

Reply:
Done Thanks!